packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -444,9 +449,24 @@ export class MultiSinkLogger extends TelemetryLogger {
 	constructor(
 		namespace?: string,
 		properties?: ITelemetryLoggerPropertyBags,
-		protected loggers: ITelemetryBaseLogger[] = [],
+		loggers: ITelemetryBaseLogger[] = [],
+		tryInheritProperties?: true,
 	) {
-		super(namespace, properties);
+		let realProperties = properties;
+		if (tryInheritProperties === true) {
+			const merge = (realProperties ??= {});
+			loggers
+				.filter((l): l is this => l instanceof TelemetryLogger)
+				.map((l) => l.properties ?? {})
+				.forEach((cv) => {
+					Object.keys(cv).forEach((k) => {
+						merge[k] = { ...cv[k], ...merge?.[k] };
+					});
+				});
+		}
```

Comment:
I was a bit confused by this bit.

- Can we avoid the `merge` variable altogether by doing `const realProperties = properties ?? {};` above? It'd also make it clearer that below we're using something that we in fact updated. However, aren't we mutating `properties` for the caller either way (if they were not undefined)? Seems like we shouldn't.
- `merge[k] = { ...cv[k], ...merge?.[k] };` initially felt like it would mess up keys with primitive values by converting them to an object, but the syntax wouldn't even produce a valid object/dictionary... so this actually works? I'd maybe add some comments to this code, and tests (for correct processing and to ensure no mutation for the caller).

Reply:
the merge var helps will type safety. ts doesn't realize after  `realProperties ??= {}` realProperties won't be undefined, as it could be mutated again, putting in a new var fixes that.

I've removed the mutation of the passed in object.


This structure is a bit different, it looks like ```
export interface ITelemetryLoggerPropertyBags {
	all?: ITelemetryLoggerPropertyBag;
	error?: ITelemetryLoggerPropertyBag;
}
```
so this is merging the bags of properties. i will add a unit test